src/Fixer/FunctionNotation/PhpdocToReturnTypeFixer.php

```diff
@@ -0,0 +1,307 @@
+            [T_STRING, '__construct'],
+            [T_STRING, '__destruct'],
+            [T_STRING, '__clone'],
+        ];
+
+        static $versionSpecificTypes = [
+            'void' => 70100,
+            'iterable' => 70100,
+            'object' => 70200,
+        ];
+
+        static $scalarTypes = [
+            'bool' => true,
+            'float' => true,
+            'int' => true,
+            'string' => true,
+        ];
+
+        static $skippedTypes = [
+            'mixed' => true,
+            'resource' => true,
+            'null' => true,
+        ];
+
+        static $classRegex = '/^\\\\?[a-zA-Z_\\x7f-\\xff](?:\\\\?[a-zA-Z0-9_\\x7f-\\xff]+)*$/';
+
+        for ($index = $tokens->count() - 1; 0 < $index; --$index) {
+            if (!$tokens[$index]->isGivenKind(T_FUNCTION)) {
+                continue;
+            }
+
+            $funcName = $tokens->getNextMeaningfulToken($index);
+            if ($tokens[$funcName]->equalsAny($blacklistFuncNames, false)) {
+                continue;
+            }
+
+            $returnTypeAnnotation = $this->findReturnAnnotations($tokens, $index);
+            if (1 !== count($returnTypeAnnotation)) {
+                continue;
+            }
+            $returnTypeAnnotation = current($returnTypeAnnotation);
+            $types = array_values($returnTypeAnnotation->getTypes());
+            $typesCount = count($types);
+            if (1 > $typesCount || 2 < $typesCount) {
+                continue;
+            }
+
+            $isNullable = false;
+            $returnType = current($types);
+            if (2 === $typesCount) {
+                $null = $types[0];
+                $returnType = $types[1];
+                if ('null' !== $null) {
+                    $null = $types[1];
+                    $returnType = $types[0];
+                }
+
+                if ('null' !== $null) {
+                    continue;
+                }
+
+                $isNullable = true;
+
+                if (PHP_VERSION_ID < 70100) {
+                    continue;
+                }
+
+                if ('void' === $returnType) {
+                    continue;
+                }
+            }
+
+            if ('static' === $returnType) {
+                $returnType = 'self';
+            }
+
+            if (isset($skippedTypes[$returnType])) {
+                continue;
+            }
+
+            if (isset($versionSpecificTypes[$returnType]) && PHP_VERSION_ID < $versionSpecificTypes[$returnType]) {
+                continue;
+            }
+
+            if (isset($scalarTypes[$returnType]) && false === $this->configuration['scalar_types']) {
+                continue;
+            }
+
+            if (1 !== Preg::match($classRegex, $returnType)) {
+                continue;
+            }
+
+            $startIndex = $tokens->getNextTokenOfKind($index, ['{', ';']);
+
+            if ($this->hasReturnTypeHint($tokens, $startIndex)) {
+                continue;
+            }
```

Comment:
For the records, segmented `if () { continue; }` are intended: force a clear mapping between feature/bug and code-coverage.